### ISSUE TITLE
Generated SubscriptionExecutor doesn't compile

### ISSUE THREAD

**Issue by beepsoft** [graphql-java-generator/graphql-maven-plugin-project#54]
Hi,

Using this graphql schema:

```graphql

scalar DateTime

input DateTimeFilter {
    eq: DateTime
    le: DateTime
    lt: DateTime
    ge: DateTime
    gt: DateTime
}

type DEntry {
    id: ID!
    type: String
    createdAt: DateTime
    updatedAt: DateTime
    deletedAt: DateTime
    license: String
    source: String
    sourceUrl: String
    sourceId: String
    translationExcerpt: String
}

input DEntryFilter {
    id: [ID!]
    createdAt: DateTimeFilter
    updatedAt: DateTimeFilter
    deletedAt: DateTimeFilter
    and: DEntryFilter
    or: DEntryFilter
    not: DEntryFilter
}

type Subscription {
    getDEntry(id: ID!): DEntry
    queryDEntry(filter: DEntryFilter): [DEntry]
}
```

When running

```
mvn clean compile
```

I get such errors:

```
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 4.028 s
[INFO] Finished at: 2020-12-05T11:29:41+01:00
[INFO] Final Memory: 35M/134M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.5.1:compile (java-compile) on project graphql-maven-bug: Compilation failure: Compilation failure: 
[ERROR] /Users/balazs/Documents/work/sztaki/szotar/sztakidict2020/trials/graphql-maven-bug/target/generated-sources/graphql-maven-plugin/test/client/util/SubscriptionExecutor.java:[611,56] no suitable method found for execute(com.graphql_java_generator.client.request.ObjectResponse,java.util.Map<java.lang.String,java.lang.Object>,com.graphql_java_generator.client.SubscriptionCallback<java.util.List<test.client.DEntry>>,java.lang.String,java.lang.Class<test.client.Subscription>,java.lang.Class<test.client.DEntry>)
[ERROR]     method com.graphql_java_generator.client.QueryExecutor.<R>execute(com.graphql_java_generator.client.request.AbstractGraphQLRequest,java.util.Map<java.lang.String,java.lang.Object>,java.lang.Class<R>) is not applicable
[ERROR]       (cannot infer type-variable(s) R
[ERROR]         (actual and formal argument lists differ in length))
[ERROR]     method com.graphql_java_generator.client.QueryExecutor.<R,T>execute(com.graphql_java_generator.client.request.AbstractGraphQLRequest,java.util.Map<java.lang.String,java.lang.Object>,com.graphql_java_generator.client.SubscriptionCallback<T>,java.lang.String,java.lang.Class<R>,java.lang.Class<T>) is not applicable
[ERROR]       (inference variable T has incompatible equality constraints test.client.DEntry,java.util.List<test.client.DEntry>)
[ERROR] /Users/balazs/Documents/work/sztaki/szotar/sztakidict2020/trials/graphql-maven-bug/target/generated-sources/graphql-maven-plugin/test/client/util/SubscriptionExecutor.java:[694,56] no suitable method found for execute(com.graphql_java_generator.client.request.ObjectResponse,java.util.Map<java.lang.String,java.lang.Object>,com.graphql_java_generator.client.SubscriptionCallback<java.util.List<test.client.DEntry>>,java.lang.String,java.lang.Class<test.client.Subscription>,java.lang.Class<test.client.DEntry>)
[ERROR]     method com.graphql_java_generator.client.QueryExecutor.<R>execute(com.graphql_java_generator.client.request.AbstractGraphQLRequest,java.util.Map<java.lang.String,java.lang.Object>,java.lang.Class<R>) is not applicable
[ERROR]       (cannot infer type-variable(s) R
[ERROR]         (actual and formal argument lists differ in length))
[ERROR]     method com.graphql_java_generator.client.QueryExecutor.<R,T>execute(com.graphql_java_generator.client.request.AbstractGraphQLRequest,java.util.Map<java.lang.String,java.lang.Object>,com.graphql_java_generator.client.SubscriptionCallback<T>,java.lang.String,java.lang.Class<R>,java.lang.Class<T>) is not applicable
[ERROR]       (inference variable T has incompatible equality constraints test.client.DEntry,java.util.List<test.client.DEntry>)
```

Here's my repo demonstrating this error:

https://github.com/beepsoft/graphql-maven-bug

**Comment by etienne-sf** [comment]
Grrrrr

I've put a protection for such cases. But "and", "or" and "not" are not in the list of keywords know by the plugin!
:(

Where did I take this f.... list ???

I'll correct that and deliver a new version
Etienne

**Comment by beepsoft** [comment]
No problem 😄 Thank you for your efforts maintaining this plugin!

**Comment by beepsoft** [comment]
By the way, the issue doesn't seem to be related to the input fields named `and, or, not`. If I remove those the compilation error still happens.

**Comment by beepsoft** [comment]
Also, not actually related to this issue, but is there an example of how subscriptions work?

For me it seems the way it is implemented in graphql-java-generator doesn't follow the protocol used by eg. Dgraph, Hasura, Apollo, etc.:

https://github.com/apollographql/subscriptions-transport-ws/blob/master/PROTOCOL.md

**Comment by etienne-sf** [comment]
Hum, it was closed by my commit comment. 
So I reopen it. 

For a sample, you can take a look at the allGraphQLCases, in the project.  
I just add a subscription package to show the usage of this.  
I’ll check the site documentation.

**Comment by etienne-sf** [comment]
> 
> 
> By the way, the issue doesn't seem to be related to the input fields named `and, or, not`. If I remove those the compilation error still happens.

Yes, that's right. It's the reason why I changed my initial comment.

The issue was a quick complex one, due to the Java generic types. It was quite tricky to solve.
But stackoverflow has a lot of solutions!
:)

About the respect of protocol : as the server is handled by graphql-java, I guess it respects plainly the GraphQL protocol.
BTW, I tested it only agains a GraphQL server powered by graphql-java.

If there are issues there, I'll be glad to test it against other kind of servers.



And about a doc on how to use it:
The plugin web site has a page about this: [https://graphql-maven-plugin-project.graphql-java-generator.com/client_subscription.html](https://graphql-maven-plugin-project.graphql-java-generator.com/client_subscription.html)

You'll find a sample in the project itself: [https://github.com/graphql-java-generator/graphql-maven-plugin-project/tree/master/graphql-maven-plugin-samples/graphql-maven-plugin-samples-allGraphQLCases-client/src/main/java/org/allGraphQLCases/subscription](https://github.com/graphql-java-generator/graphql-maven-plugin-project/tree/master/graphql-maven-plugin-samples/graphql-maven-plugin-samples-allGraphQLCases-client/src/main/java/org/allGraphQLCases/subscription)

Etienne

**Comment by etienne-sf** [comment]
I just released the 1.11.1 version.

Please tell me if everything is ok for you, now.

Etienne

**Comment by beepsoft** [comment]
@etienne-sf thank you and stackoverflow so much! 😃 It compiles all right now! I really appreciate your quick response to this issue!

I will take a look at the examples as well, thanks. 

I guess graphql-java only provides the websocket layer and doesn't bother with the message format sent over the line. But I will investigate it a bit too to see if I can make it work. You may also try to create a small [Hasura](https://hasura.io/) setup and see whether graphql-java-generator subscriptions work with it.
